docs/api/v3.rst

```diff
@@ -0,0 +1,320 @@
+with a specific ``Token`` that provide access to different resources.
+
+Available scopes
+++++++++++++++++
+
+* ``project``
+  * ``project:admin``: access to change admin settings (similar to Admin tab)
+  * ``project:read``: access to read all the information about the projects
+  * ``project:write``: access to anything related to perform write actions (triggering a Build for this project, activate/deactivate versions, etc)
+  * ``project:docs``: access any file from any version of the project's documentation
+
+
+.. TODO:
+
+   We could extend the scopes to be per-project or per-user. I'm not
+   sure if this is possible yet, but we should consider it.
+
+   * per-project: the user could go to the Project Admin tab and create a Token for this specific project
+   * per-user: the user could go to their own User settings and create token for all his projects
+
+
+Resources
+---------
+
+Projects
+~~~~~~~~
+
+Project list
+++++++++++++
+
+.. http:get::  /api/v3/project/
+
+    Retrieve a list of all the projects of the current logged in user.
+
+    **Example request**:
+
+    .. sourcecode:: bash
+
+        $ curl https://readthedocs.org/api/v3/project/
+
+    **Example response**:
+
+    .. sourcecode:: js
+
+        {
+            "count": 25,
+            "next": "/api/v3/project/limit=10&offset=10",
+            "previous": null,
+            "results": [PROJECTS]
+        }
+
+    :>json integer count: Total number of Projects.
+    :>json string next: URI for next set of Projects.
+    :>json string previous: URI for previous set of Projects.
+    :>json array results: Array of ``Project`` objects.
+
+.. TODO:
+
+   Add query string filters to narrow the query:
+     * privacy level
+     * language
+     * programming language
+     * repo url
+     * repo type
+     * version active
+     * version built
+     * all database?
+
+
+Project details
++++++++++++++++
+
+.. http:get::  /api/v3/project/(string:slug)/
+
+    Retrieve details of a single project.
+
+    .. sourcecode:: js
+
+        {
+            "name": "Pip",
+            "slug": "pip",
+            "description": "Pip Installs Packages.",
+            "language": "en",
+            "programming_language": "py",
+            "repo": "https://github.com/pypa/pip",
+            "repo_type": "git",
+            "default_version": "stable",
+            "default_branch": "master",
+            "documentation_type": "sphinx_htmldir",
+            "canonical_url": "http://pip.pypa.io/en/stable/",
+            "links": [
+                {
+                    "href": "/api/v3/project/pip/users/",
+                    "rel": "users",
+                    "type": "GET"
+                },
+                {
+                    "href": "/api/v3/project/pip/versions/",
+                    "rel": "versions",
+                    "type": "GET"
+                },
+                {
+                    "href": "/api/v3/project/pip/builds/",
```

Comment:
I'm thinking that you can do a POST to this URL with something like:

```
{
  "version": "latest"
}
```

and trigger a build for that project and version.

Reply:
and probably similar POST request for the rest of the endpoints listed here as `links` and GET.

`/domains` could add a new domain, `/notifications` a new webhook or email notification, etc